src/lizaalert/courses/views.py

```diff
@@ -236,6 +237,27 @@ def unroll(self, request, **kwargs):
         subscription.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @swagger_auto_schema(
+        responses={
+            status.HTTP_200_OK: "Курс успешно завершен.",
+        }
+    )
+    @action(detail=True, methods=["post"], permission_classes=(IsAuthenticated, EnrolledAndCourseHasStarted))
+    def complete(self, request, **kwargs):
+        """
+        Завершить курс для пользователя.
+
+        Данное событие вызывает event завершения курса.
+        """
+        user = self.request.user
+        try:
+            course = get_object_or_404(Course, **kwargs)
+        except ValueError:
+            raise BadRequestException({"detail": "Invalid id."})
+        course.finish(user)
+        course_completion_event(course, user)
```

Comment:
```suggestion
        after_course_completion_event(course, user)
```

давай это сделаем на сигналах

Reply:
https://docs.djangoproject.com/en/5.0/topics/signals/